### PR TITLE
Account for Custody and Liabilities in Spot Prices

### PR DESCRIPTION
**Pull request by AustinoBombino**

### CODE REVIEW

File: x/clp/keeper/calculations.go [hunk 1]
```diff
@@ -354,8 +364,8 @@ func CalcSpotPriceX(X, Y sdk.Uint, decimalsX, decimalsY uint8, pmtpCurrentRunnin
 
 	return RatToDec(&pmtpPrice)
 }
-func CalcRowanValue(pool *types.Pool, pmtpCurrentRunningRate sdk.Dec, rowanAmount sdk.Uint) (sdk.Uint, error) {
-	spotPrice, err := CalcRowanSpotPrice(pool, pmtpCurrentRunningRate)
+func CalcRowanValue(pool *types.Pool, pmtpCurrentRunningRate sdk.Dec, rowanAmount sdk.Uint, marginEnabled bool) (sdk.Uint, error) {
```
Comment (AustinoBombino) [inline x/clp/keeper/calculations.go:367]: @canercandan @Jedi2002 not 100% sure about the usage of this function and whether we want this accounting present for this one

**Comment by canercandan** [comment]
@AustinoBombino swap prices:
```
pool:
  external_asset:
    symbol: cfrax
  external_asset_balance: "239850471300746838222957"
  external_custody: "0"
  external_liabilities: "0"
  health: "1.000000000000000000"
  interest_rate: "0.000000022000000000"
  last_height_interest_rate_computed: "161413"
  native_asset_balance: "24701699854410803979303000"
  native_custody: "0"
  native_liabilities: "0"
  pool_units: "31851103132610755449464526"
  reward_period_native_distributed: "130164962496649937516"
  swap_price_external: "102.988264757263981347"
  swap_price_native: "0.009709844149301174"

```

**Review by canercandan — DISMISSED** [review]
Lgtm

**Review by canercandan — APPROVED** [review]

**Comment by canercandan** [comment]
open 1000 USDC with 10x leverage against Rowan

Pool State:
```
pools:
- external_asset:
    symbol: cusdc
  external_asset_balance: "611316557165"
  external_custody: "0"
  external_liabilities: "9000000000"
  health: "0.985491278773643856"
  interest_rate: "0.000000010999999999"
  last_height_interest_rate_computed: "15"
  native_asset_balance: "79978669084012644657289180"
  native_custody: "1306450091106295598863316"
  native_liabilities: "0"
  pool_units: "81285119183944010790286697"
  reward_period_native_distributed: "0"
  swap_price_external: "131.038125995881888844"
  swap_price_native: "0.007631366767496559"
```

verify swap price native:
```
((611316557165+9000000000)÷10^6)÷((79978669084012644657289180+1306450091106295598863316)÷10^18)
= 0.0076313668
```

verify swap price external:
```
1 / 0.0076313668
131.0381259959
```

**Pull request merged**